### PR TITLE
Use short hash names to identify references in commit history

### PR DESCRIPTION
**Pull request by cbrgm**
This PR allows you to use the 7 first chars of the commit hash to identify references.

So instead of
```
gchl since a74d9491c1b6fc571ac6e1d0146cbbb45fd01b63
gchl between  a74d9491c1b6fc571ac6e1d0146cbbb45fd01b63 6b8ca2f143a6b1372db42fbd5f7168f5d6d50302

```

you can now use
```
gchl since a74d949
gchl since a74d949 6b8ca2f
```
`vendors/*` has been updated.
This PR also includes some code refactoring.

Cheers

### CODE REVIEW

File: pkg/git/git.go [hunk 1]
```diff
@@ -60,8 +62,27 @@ func (g *Git) getHashObject(hash string) (*plumbing.Reference, error) {
 		return nil, err
 	}
 
-	reference := plumbing.NewReferenceFromStrings(hash, hash)
-	return reference, nil
+	result := plumbing.NewReferenceFromStrings(hash, hash)
+	return result, nil
+}
+
+func (g *Git) getHashObjectByPrefix(hash string) (*plumbing.Reference, error) {
```
Comment (kdomanski) [inline pkg/git/git.go:75]: I think it's better to continue here and fail if another match is found - thus it won't just return the first match when prefixes collide. As an added bonus, you'll be able to make it use prefixes of any length.

Comment (kdomanski) [reply pkg/git/git.go:75]:
```
if len(hash) > 40 {
  return nil, fmt.Errorf("hash prefix %s is too long", hash)
}
```

File: pkg/git/git.go [hunk 2]
```diff
@@ -60,8 +62,27 @@ func (g *Git) getHashObject(hash string) (*plumbing.Reference, error) {
...
+	commits, err := g.repo.CommitObjects()
+	if err != nil {
+		return nil, err
+	}
+
+	var result *plumbing.Reference
+	commits.ForEach(func(reference *object.Commit) error {
+		commitHash := reference.Hash.String()[0:7]
```
Comment (kdomanski) [inline pkg/git/git.go]:
```
commitHash := reference.Hash.String()[0:len(hash)]
```

File: pkg/git/git.go [hunk 3]
```diff
@@ -60,8 +62,27 @@ func (g *Git) getHashObject(hash string) (*plumbing.Reference, error) {
...
+		if commitHash == hash {
```
Comment (kdomanski) [inline pkg/git/git.go]:
```
if commitHash == hash {
  if result != nil {
    return fmt.Errorf("ambiguous hash prefix %s: %s and %s match", hash, result.Hash.String(), reference.Hash.String())
  }
  result = plumbing.NewReferenceFromStrings(reference.Hash.String(), reference.Hash.String())
}
```
And preserve the error checking of `ForEach`

**Comment by cbrgm** [comment]
@kdomanski Thanks for your review. Did some rework. I improved the error handling and references can now be found using a hash of any length. `gchl` will cause an error if multiple references are found.

**Review by kdomanski — APPROVED** [review]
Nice. I just generated a log from `v2.6.0` to `ccf` ;-)

**Pull request merged**